src/rgw/rgw_admin.cc

```diff
@@ -8681,15 +8709,26 @@ int main(int argc, const char **argv)
   }
 
   if (opt_cmd == OPT::DATALOG_TRIM) {
-    utime_t start_time, end_time;
-
-    int ret = parse_date_str(start_date, start_time);
-    if (ret < 0)
-      return -ret;
-
-    ret = parse_date_str(end_date, end_time);
-    if (ret < 0)
-      return -ret;
+    if (!start_date.empty()) {
+      std::cerr << "start-date not allowed." << std::endl;
+      return -EINVAL;
+    }
+    if (!end_date.empty()) {
+      std::cerr << "end-date not allowed." << std::endl;
+      return -EINVAL;
+    }
+    if (!end_marker.empty()) {
+      std::cerr << "end-marker not allowed." << std::endl;
+      return -EINVAL;
+    }
+    if (!end_marker.empty()) {
+      if (marker.empty()) {
+	marker = end_marker;
+      } else {
+	std::cerr << "start-marker and marker not both allowed." << std::endl;
+	return -EINVAL;
+      }
+    }
```

Comment:
this conflates start_marker and end_marker. for cls_fifo, trim can only support end_marker. the first block should check for `!start_marker.empty()` and complain that "start-marker is not allowed". the second block should print "end-marker and marker not both allowed"

the MDLOG_TRIM changes have the same issue

Reply:
Done.